static/app/views/profiling/index.tsx

```diff
@@ -1,23 +1,96 @@
+import {useCallback, useEffect, useState} from 'react';
+import * as Sentry from '@sentry/react';
+
+import {PromptData, promptsCheck, promptsUpdate} from 'sentry/actionCreators/prompts';
 import Feature from 'sentry/components/acl/feature';
 import Alert from 'sentry/components/alert';
+import LoadingIndicator from 'sentry/components/loadingIndicator';
 import {t} from 'sentry/locale';
 import {PageContent} from 'sentry/styles/organization';
-import {Organization} from 'sentry/types';
+import {Organization, RequestState} from 'sentry/types';
+import trackAdvancedAnalyticsEvent from 'sentry/utils/analytics/trackAdvancedAnalyticsEvent';
+import useApi from 'sentry/utils/useApi';
 import withOrganization from 'sentry/utils/withOrganization';
 
+import ProfilingOnboarding from './profilingOnboarding';
+
+function renderNoAccess() {
+  return (
+    <PageContent>
+      <Alert type="warning">{t("You don't have access to this feature")}</Alert>
+    </PageContent>
+  );
+}
+
+function shouldShowProfilingOnboarding(state: RequestState<PromptData>): boolean {
+  if (state.type === 'resolved') {
+    return typeof state.data?.dismissedTime !== 'number';
+  }
+
+  return false;
+}
+
 type Props = {
   children: React.ReactChildren;
   organization: Organization;
 };
 
 function ProfilingContainer({organization, children}: Props) {
-  function renderNoAccess() {
-    return (
-      <PageContent>
-        <Alert type="warning">{t("You don't have access to this feature")}</Alert>
-      </PageContent>
-    );
-  }
+  const client = useApi();
+
+  const [requestState, setRequestState] = useState<RequestState<PromptData>>({
+    type: 'initial',
+  });
+
+  // Fetch prompt data and see if we need to show the onboarding.
+  useEffect(() => {
+    setRequestState({type: 'loading'});
+    promptsCheck(client, {
+      organizationId: organization.id,
+      feature: 'profiling_onboarding',
+    })
+      .then(data => {
+        setRequestState({type: 'resolved', data});
+      })
+      .catch(e => {
+        Sentry.captureException(e);
+        setRequestState({type: 'errored', error: t('Error: Unable to load prompt data')});
+      });
+  }, [client, organization]);
+
+  // Eagerly update state and update check
+  const dismissPrompt = useCallback(
+    (status: 'done' | 'dismissed') => {
+      setRequestState({type: 'resolved', data: {dismissedTime: Date.now()}});
+      trackAdvancedAnalyticsEvent(
+        status === 'done'
+          ? 'profiling_views.onboarding_action.done'
+          : 'profiling_views.onboarding_action.dismiss',
+        {
+          action: status === 'done' ? 'done' : 'dismiss',
```

Comment:
nit: this ternary isn't necessary
```suggestion
          action: status,
```

Reply:
can we align `status` and `action` ? right now `action` is 'done | dismiss' but status is 'done | dismiss*ed*'?